meeting-notes/2018/2018-12-05--gui-and-in-web-browsers-weekly.md

```diff
@@ -0,0 +1,130 @@
+
+- @lidel
+	- Done:
+		- Spent big chunk of time on reviewing IPFS and libp2p Roadmaps, added some comments and questions
+		- Provided some knowledge transfer about WebExtension stores to Discussify team :)
+		- [RFC: Support ArrayBuffer as an alternative to Buffer input argument](https://github.com/ipfs/interface-ipfs-core/issues/404)
+		- Added ipfs-share-files to Transifex: https://www.transifex.com/ipfs/ipfs-share-files/
+		- Origin Trial for Signed HTTP Exchanges is enabled for ipfs.io Gateway – details in https://github.com/ipfs/infrastructure/issues/453#issuecomment-442944274 & https://github.com/ipfs/in-web-browsers/issues/121#issuecomment-442952118
+		- Fixed local build of blog.ipfs.io - https://github.com/ipfs/blog/pull/199
+		- Fixed ipfs-companion’s API access in Chrome 72 - https://github.com/ipfs-shipyard/ipfs-companion/pull/631
+		- Confirmed js-ipfs repo can be shared by multiple pages when embedded via iframe
+	- Blocked:
+		- Waiting for cert from Digicert (for SXG experimentation)
+		- Companion fails to build after dependency bump, looking into it (webpack/terser)
+	- Next:
+		- Release new Companion Beta release with fix for Chrome Beta
+		- Blog post about Signed HTTP Exchanges at ipfs.io
+		- Finish capabilities preload via window.ipfs.enable and ship to Beta
+
+
+- @terichadbourne
+	- Done:
+		- Opened an issue proposing a workshop using the file API (rather than DAG API) for ProtoSchool. (All current workshops use the DAG API so it’s not being surfaced that file transfer is possible.) Open to suggestions and feedback at https://github.com/ipfs-shipyard/proto.school/issues/91
+		- Was guinea pig in usability test of Web UI with Oli
+		- ProtoSchool updated to use the new CID link API (about to be merged from https://github.com/ipfs-shipyard/proto.school/pull/88)
+		- Taught ProtoSchool to save state of code in editor box and completion status of each lesson (to be merged this week from<https://github.com/ipfs-shipyard/proto.school/pull/86>)
+		- Conducted usability test of ProtoSchool
+
+
+- @olizlla
+	- Done:
+		- Short week, was off thurs / friday
+		- 3x Useability interviews for Web UI - https://github.com/ipfs-shipyard/ipfs-webui/tree/master/docs/usability/notes
+		- cleanup repository before starting up IPFS PR reviews
+		- Kick off “add Desktop to ipfs.io website” https://github.com/ipfs/website/issues/281 to honour it’s upcoming new release.
+		- Get specton test working for ipfs desktop - https://github.com/ipfs-shipyard/ipfs-desktop/pull/730
+		- Docs PR review https://github.com/ipfs/docs/pull/67
+	- Blocked:
+		- 	N/A
+	- Next:
+		- 2 more usability testing session for webui. write up the common issues. Propose next round of improvements to webui
+		- Replace distribution of peers graph in webui with repo size information. Limit geolocations to peers page to avoid gratuitous system load.
+		- Cut a release of Web UI
+		- PR go-ipfs to add webui link to ipfs daemon output.
+		- GET DESKTOP RELEASED 🚢 ✨
```

Comment:
Did this happen?

Reply:
No. Most of the time available has gone on
- Dealing with errors where a previous ifps deaemon wasn't shutdown properly https://github.com/ipfs-shipyard/ipfs-desktop/pull/722 - this ment the app would fail to start if the user previously ran `ipfs daemon` manually, and didn't wait for the process to shutdown gracefully when exiting.
- Dealing with issues related to CORS that was raised during reviews of the unclean startup PR.  We were initially editing the users node config to include a CORS origin for webui://- which is the fake scheme under which we serve the webui from within the electron app. We then decided that we didn't want to leave the user with a node config that included a CORS origin that would live on beyond the life of the app, as we can't predict how it could be abused in the future. 
  - So @hacdias refactored to [proxying ipfs](https://github.com/ipfs-shipyard/ipfs-desktop/pull/738) but a quick test showed that it crashed the electron webview on large files. This is probably fixable, but we decided to keep things simple and disable CORS checks for the webview, as we are in control of the window and the webui code. As a precuation we are enabling the other electron security recommendations in https://github.com/ipfs-shipyard/ipfs-desktop/pull/741

I'd like to have a pre-release this afternoon, though it looks like it will have to be tomorrow now, and get it installed on gui team machines, so we can test out the auto-update mechanism, and then invite the IPFS org to give it a go.

Reply:
Thank you for the super complete status update. ❤️ @olizilla